wgsl/index.bs

```diff
@@ -534,7 +534,191 @@ When unclear from context, this specification indicates
+    * If *NS* is `off`, then *D* is discarded (i.e. deleted, or forgotten).
+    * Otherwise, *NS* is either `error` or `warning`.
+        Set *D*'s [=diagnostic/severity=] property to *NS*.
+* Otherwise, *D* is unchanged.
+
+A <dfn noexport>range diagnostic filter</dfn> is a [=diagnostic filter=] that is applied to each
+diagnostic whose [=diagnostic/triggering location=] falls within a specified range of source text,
+and when no other more specific filter applies.
+A range diagnostic filter is specified as a `@diagnostic` [=attribute=] immediately before the affected source range,
+as specified in the following table.  A `@diagnostic` attribute [=shader-creation error|must not=] appear anywhere else.
+
+<table class='data'>
+  <caption>Placement of a range diagnostic filter</caption>
+  <thead>
+    <tr><th>Placement<th>Affected Range
+  </thead>
+    <tr><td>Beginning of a [=compound statement=].<td>The compound statement.
+    <tr><td>Beginning of a [=function declaration=].<td>The function declaration.
+    <tr><td>Beginning of a [=statement/if=] statement.<td>The if statement: the [=syntax/if_clause=]
+      and all associated [=syntax/else_if_clause=] and [=syntax/else_clause=] clauses,
+      including all controlling condition expressions.
+    <tr><td>Beginning of a [=statement/switch=] statement.<td>The switch statement: the selector expression and the [=syntax/switch_body=].
+    <tr><td>Beginning of a [=syntax/switch_body=].<td>The [=syntax/switch_body=].
+    <tr><td>Beginning of a [=statement/loop=] statement.<td>The loop statement.
+    <tr><td>Beginning of a [=statement/while=] statement.<td>The while statement: both the condition expression and the loop body.
+    <tr><td>Beginning of a [=statement/for=] statement.<td>The for statement: the [=syntax/for_header=] and the loop body.
+    <tr><td>Immediately before the opening brace (`'{'`) of the [=loop body=] of a [=statement/loop=], [=statement/while=], or [=statement/for=] loop.<td>The [=loop body=].
+    <tr><td>Beginning of a [=syntax/continuing_compound_statement=].<td>The [=syntax/continuing_compound_statement=].
+</table>
+
+Note: A [=function body=], a [=case clause=], a [=default-alone clause=],
+and the bodies of [=syntax/else_if_clause=] and [=syntax/else_clause=] are themselves [=compound statements=].
+
+<div class='example wgsl global-scope' heading='Range diagnostic filter on texture sampling'>
+  <xmp highlight='rust'>
+  var<private> d: f32;
+  fn helper() -> vec4<f32> {
+    // Disable the derivative_uniformity diagnostic in the
+    // body of the "if".
+    if (d < 0.5) @diagnostic(off,derivative_uniformity) {
+      return textureSample(t,s,vec2(0,0));
+    }
+    return vec4(0.0);
+  }
+  </xmp>
+</div>
+
+A <dfn noexport>global diagnostic filter</dfn> is a diagnostic filter whose [=affected range=] is the whole WGSL program.
+It is a [=directive=], thus appearing before any [=module scope|module-scope=] declarations.
```

Comment:
So `enable` and `diagnostic` directives can be declared in any order, or mixed, correct?
Would be nice to demonstrate this in the examples below, except the spec doesn't have any real extensions defined yet. Never mind.

Reply:
Yes, they can be intermixed.
A reader can infer that from the translation_unit and global_directive grammar rules.